### PR TITLE
test cases for PAC

### PR DESCRIPTION
**Pull request by VeereshAradhya**
Added test cases for Pipelines As Code along with test data

Jira Id: [ODC-5149](https://issues.redhat.com/browse/ODC-5149)

Acceptance Criteria: 

1. As a user, looking at the Pipelines page in the Developer Console, I should be able to see a list of (a) Git repositories that are added to the namespace for PAC execution AND (b) all pipelines in the namespace
2. As a user, I should be able to navigate to a details page of the git repo.
    1. This details page should provide access to (a) details of the git repo and (b) a list of pipeline runs.
    2. This PLR tab should show additional information than the typical PLR List view, including SHA (commit id), commit message, branch & trigger type
3. As a user, when looking at a Pipeline Run Details page, if associate with a git repo (PAC),
4. Indicate that it's from a specific git repo rather than a PL resource
5. Include the SHA (commit id), commit message, branch & trigger type

Pre-conditions for Epic validation:

1. Openshift Pipelines is installed 
2. Pipelines As Code is installed

Checks required for approving Epic gherkin scripts PR:

- [x] Add @epic-number to the scenarios or feature file [e.g: @odc-xxx]
- [x] Add @to-do for automation possible scenarios
- [x] Add @regression or @smoke based on the importance of the scenario
- [x]  Update the test scenarios count in Automation status confluence Report
- [x]  Check for the linter issues by executing yarn run gherkin-lint on frontend folder [Skip epic number tags related linter issues]

@makambalaji @sanketpathak, please review

### CODE REVIEW

File: frontend/packages/pipelines-plugin/integration-tests/features/pipelines/pipelines-as-code.feature [hunk 1]
```diff
@@ -0,0 +1,83 @@
+@pipelines
+Feature: Create the repo
+              As a developer, I want to create, edit, delete and view the repositories
+        Background:
+            Given user has created or selected namespace "aut-pipelines"
+            Given user has installed pipelines as code
+        
+        @smoke @pre-condition
```
Comment (makambalaji) [inline frontend/packages/pipelines-plugin/integration-tests/features/pipelines/pipelines-as-code.feature:8]: @smoke is not required, As @pre-condition tagged scripts will execute by default

File: frontend/packages/pipelines-plugin/integration-tests/features/pipelines/pipelines-as-code.feature [hunk 2]
```diff
@@ -0,0 +1,83 @@
+@pipelines
+Feature: Create the repo
+              As a developer, I want to create, edit, delete and view the repositories
+        Background:
+            Given user has created or selected namespace "aut-pipelines"
+            Given user has installed pipelines as code
```
Comment (makambalaji) [inline frontend/packages/pipelines-plugin/integration-tests/features/pipelines/pipelines-as-code.feature:6]:
```suggestion
            And user has installed pipelines as code
```

File: frontend/packages/pipelines-plugin/integration-tests/features/pipelines/pipelines-as-code.feature [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+        
+        @smoke @pre-condition
+        Scenario: Repositories page: P-11-TC01
```
Comment (makambalaji) [inline frontend/packages/pipelines-plugin/integration-tests/features/pipelines/pipelines-as-code.feature:9]:
```suggestion
        Scenario Outline: Repositories page: P-11-TC01
```

File: frontend/packages/pipelines-plugin/integration-tests/features/pipelines/pipelines-as-code.feature [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+              
+        Examples:
+                  | repository_name |
+                  | test-repo       |
+        
+        @smoke
+        Scenario: Kebab menu options of newly created repository in Repositories page: P-11-TC04
+            Given repository "test-repository" is present on Repositories page
+             When user searches repository "test-repository" in repsitories page
+              And user clicks on kebab menu of the repository "test-repository"
```
Comment (makambalaji) [inline frontend/packages/pipelines-plugin/integration-tests/features/pipelines/pipelines-as-code.feature:51]: I think repository name is "test-repo"

File: frontend/packages/pipelines-plugin/integration-tests/features/pipelines/pipelines-as-code.feature [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+             Then kebab menu displays with options Edit labels, Edit annotations, Edit repository, Delete repository
+        
+        @smoke
+        Scenario: Actions menu of newly created reposiotry in Repository details page: P-11-TC05
+            Given useris at repositories details page with newly created repository "test-repo"
```
Comment (makambalaji) [inline frontend/packages/pipelines-plugin/integration-tests/features/pipelines/pipelines-as-code.feature:56]:
```suggestion
            Given user is at repositories details page with newly created repository "test-repo"
```

File: frontend/packages/pipelines-plugin/integration-tests/features/pipelines/pipelines-as-code.feature [hunk 6]
```diff
@@ -0,0 +1,83 @@
...
+        Examples:
+                  | repository_name |
+                  | test-repo       |
+
+        @regression
+        Scenario: Delete the Repository from repositories page: P-11-TC07
+            Given repository "test-repo" is present on Repositories page
+             When user selects "Delete Repository" from the kebab menu for "test-repo"
+              And user clicks Delete button on Delete Reporsitory modal
+             Then user will be redirected to Repositories page
```
Comment (makambalaji) [inline frontend/packages/pipelines-plugin/integration-tests/features/pipelines/pipelines-as-code.feature:79]: Same action is performed in above scenario, So IMO this scenario is not required. Instead we can edit the repository

**Review by makambalaji — COMMENTED** [review]
Please add @odc-xxx epic number on top of the feature file
Add 2 lines of space between scenarios and Background, To maintain same gherkin style

Can we can add few more scenarios related to edit repository, labels and annotations?

File: frontend/packages/pipelines-plugin/integration-tests/features/pipelines/pipelines-as-code.feature (hunk 6 above)
Comment (VeereshAradhya) [reply frontend/packages/pipelines-plugin/integration-tests/features/pipelines/pipelines-as-code.feature:79]: The only way we can edit the repository is by editing yaml content

Comment (makambalaji) [reply frontend/packages/pipelines-plugin/integration-tests/features/pipelines/pipelines-as-code.feature]: That is fine, please add it

**Comment by makambalaji** [comment]
> Please add @odc-xxx epic number on top of the feature file
> Add 2 lines of space between scenarios and Background, To maintain same gherkin style
> 
> Can we can add few more scenarios related to edit repository, labels and annotations?

Please add scenarios as mentioned above and also update the styling as per the gherkin-linter

File: frontend/packages/pipelines-plugin/integration-tests/features/pipelines/pipelines-as-code.feature [hunk 7]
```diff
@@ -0,0 +1,121 @@
...
+
+        @regression @to-do
+        Scenario: Edit label of repository: P-11-TC09
+            Given repository "test-repo" is present on the Repositories page
+             When user searches repository "test-repo" in repositories page
+              And user clicks repository "test-repo" from searched results on Repositories page
+              And user clicks on Edit button of the labels section
+              And adds the label "check=label"
+              And clicks on the Save button
+             Then label "check=label" should be present in the label section
```
Comment (makambalaji) [inline frontend/packages/pipelines-plugin/integration-tests/features/pipelines/pipelines-as-code.feature:110]:
```suggestion
             Then label "check=label" should be present in the labels section
```

File: frontend/packages/pipelines-plugin/integration-tests/features/pipelines/pipelines-as-code.feature [hunk 8]
```diff
@@ -0,0 +1,121 @@
+@pipelines @odc-5149
+Feature: Create the repo
```
Comment (makambalaji) [inline frontend/packages/pipelines-plugin/integration-tests/features/pipelines/pipelines-as-code.feature:2]:
```suggestion
Feature: Perform Actions on repository
```

**Review by makambalaji — COMMENTED** [review]
Small changes needed and 
Please update the Automation status report : https://docs.jboss.org/display/ODC/Automation+Status+Report and select the check box  in pr description

File: frontend/packages/pipelines-plugin/integration-tests/features/pipelines/pipelines-as-code.feature (hunk 6 above)
Comment (VeereshAradhya) [reply frontend/packages/pipelines-plugin/integration-tests/features/pipelines/pipelines-as-code.feature]: Added

**Pull request merged**